src/main/scala/cromwell/engine/CallExecutionActor.scala

```diff
@@ -1,42 +1,70 @@
-import cromwell.engine.backend.Backend
+import cromwell.engine.backend.{TaskAbortedException, Backend}
 
 import scala.language.postfixOps
-import scala.util.{Failure, Success}
-
+import scala.util.{Success, Failure}
 
 object CallExecutionActor {
-  def props(backend: Backend, command: String, workflowDescriptor: WorkflowDescriptor, call: Call, callInputs: CallInputs, lookup: ScopedLookupFunction): Props =
-    Props(new CallExecutionActor(backend, command, workflowDescriptor, call, callInputs, lookup))
+
+  sealed trait CallExecutionActorMessage
+  case class Execute(workflowId: WorkflowID,
+                        backend: Backend,
+                        command: String,
+                        workflowDescriptor: WorkflowDescriptor,
+                        call: Call,
+                        callInputs: CallInputs,
+                        lookup: ScopedLookupFunction) extends CallExecutionActorMessage
+
+  def props(): Props = Props(new CallExecutionActor())
 }
 
 /** Actor to manage the execution of a single call. */
-class CallExecutionActor(backend: Backend, command: String, workflowDescriptor: WorkflowDescriptor, call: Call, callInputs: CallInputs, lookup: ScopedLookupFunction) extends Actor with CromwellActor {
+class CallExecutionActor() extends Actor with CromwellActor {
+  import CallExecutionActor._
+
   private val log = Logging(context.system, classOf[CallExecutionActor])
-  val tag = s"CallExecutionActor [UUID(${workflowDescriptor.shortId}):${call.name}]"
-
-  log.info(s"$tag: starting.")
-  backend.executeCommand(command, workflowDescriptor, call, callInputs, lookup) match {
-    case Success(callOutputs) =>
-      log.info(s"$tag: successful execution.")
-      context.parent ! CallActor.ExecutionFinished(callOutputs)
-    case Failure(e) =>
-      log.error(s"$tag: failed execution.")
-      context.parent ! CallActor.ExecutionFailed(e)
+  private val callExecutionActorId = UUID.randomUUID()
+
+  def tag(callReference: CallReference) = s"CallExecutionActor [UUID($callExecutionActorId)][UUID(${callReference.workflowId})/UUID(${callReference.callId})]"
+  val basicTag = s"CallExecutionActor [UUID($callExecutionActorId)]"
+
+  private def execution( workflowId: WorkflowID,
+                         backend: Backend,
+                         command: String,
+                         workflowDescriptor: WorkflowDescriptor,
+                         call: Call,
+                         callInputs: CallInputs,
+                         lookup: ScopedLookupFunction) = {
+
+    // Generate a unique CallExecution ID for this execution:
+    val callReference: CallReference = CallReference(call.callID, workflowId)
+    log.info(s"${tag(callReference)}: starting ${call.name} for workflow ${workflowDescriptor.shortId}")
+
+    val executionResult = backend.executeCommand(command, workflowDescriptor, call, callInputs, lookup, AbortFunctionRegistration(getAbortRegistrationFunction(callReference)))
```

Comment:
The logic around `AbortFunctionRegistration` is fairly complex, could you please add some comments describing how this works?